internal/config/config.go

```diff
@@ -357,3 +406,106 @@ func moduleVersionFromBuild(modulePath string) string {
 
 	return "unknown-version"
 }
+
+// Implement the refresh token source
+type workloadIdentityRefreshTokenSource struct {
+	ctx                   context.Context
+	authScope             string
+	workloadIdentityToken string
+	client                *http.Client
+	stsBaseURL            string
+}
+
+// Token implements the oauth2.TokenSource interface
+func (s *workloadIdentityRefreshTokenSource) Token() (*xoauth2.Token, error) {
+	accessToken, expiry, err := oidcTokenExchange(
+		s.ctx,
+		s.authScope,
+		s.workloadIdentityToken,
+		s.client,
+		s.stsBaseURL,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &xoauth2.Token{
+		AccessToken: accessToken,
+		Expiry:      time.Now().Add(time.Duration(expiry) * time.Second),
+	}, nil
+}
+
+func oidcTokenExchange(ctx context.Context, authScope string, workloadIdentityToken string, client *http.Client, stsBaseURL string) (string, int, error) {
+	if authScope == "" {
+		return "", 0, fmt.Errorf("Authorization scope cannot be empty for OIDC token exchange")
+	}
+
+	if workloadIdentityToken == "" {
+		return "", 0, fmt.Errorf("Workload identity token cannot be empty for OIDC token exchange")
+	}
+
+	// Construct the full token exchange URL
+	baseURL, err := url.Parse(stsBaseURL)
+	if err != nil {
+		return "", 0, fmt.Errorf("Failed to parse STS base URL: %w", err)
+	}
+	baseURL.Path = path.Join(baseURL.Path, "/use/token")
```

Comment:
There should be an OpenAPI spec for this endpoint.  Given that spec, we can onboard this service to https://github.com/equinix/equinix-sdk-go to encapsulate the request/response details & make it possible for customers to build similar integrations outside of the Terraform provider.